src/components/PopupWithConfirm.js

```diff
@@ -0,0 +1,40 @@
+import Popup from "./Popup.js";
+
+export default class PopupWithConfirm extends Popup {
+  constructor(popupSelector) {
+    super({ popupSelector });
+    this._popupForm = this._popupElement.querySelector(".modal__form");
+    this._submitButton = this._popupForm.querySelector(
+      ".form__button_type_delete"
+    );
+    // this._submitButtonText = this._submitButton.textContent;
+  }
+
+  setSubmitAction(action) {
+    console.log("aaa");
+    this._handleFormSubmit = action;
+  }
+
+  setLoading(isLoading, loadingText = "Removing...") {
+    if (isLoading) {
+      this._submitButton.textContent = loadingText;
+      this._submitButton.disabled = true;
+    } else {
+      this._submitButton.textContent = this._submitButtonText;
+      this._submitButton.disabled = false;
+    }
+  }
+
+  close() {
+    super.close();
+    this._popupForm.removeEventListener("submit", this._handleFormSubmit);
+  }
+
+  setEventListeners() {
+    super._setEventListeners();
+    this._popupForm.addEventListener("submit", (e) => {
+      e.preventDefault();
+      this._handleFormSubmit(e);
+    });
```

Comment:
If you add a listener in `setEventListeners` then you need to correctly remove it


You are not able to remove any listener in such a way.

More info here: https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/removeEventListener